app/assets/javascripts/student/interventions_controller.js

```diff
@@ -0,0 +1,194 @@
+      };
+    },
+
+    defaultSelectedIntervention: function() {
+      return (this.interventions.length === 0)
+        ? null
+        : this.interventions[0].id
+    },
+
+    // Helper for reading value from state
+    getSelectedIntervention: function() {
+      if (this.selectedInterventionId === null) {
+        return null;
+      }
+      return _.findWhere(this.interventions, { id: this.selectedInterventionId });
+    },
+
+    // This binds event listeners for user actions, and for events fired by the jquery-ujs
+    // Rails code that is adding behavior to HTML generated with `form_for` server rendering.
+    bindListeners: function () {
+      // interventions
+      this.$el.on('click', '#open-intervention-form', this.onAddNewIntervention.bind(this));
+      this.$el.on('click', '#close-intervention-form', this.onCancelNewIntervention.bind(this));
+      this.$el.on('click', '.intervention-cell', this.onSelectedIntervention.bind(this));
+      this.$el.on('ajax:success', '.new-intervention-form', this.onNewInterventionSaveSucceeded.bind(this));
+      this.$el.on('ajax:error', '.new-intervention-form', this.onNewInterventionSaveFailed.bind(this));
+
+      // progress notes
+      this.$el.on('click', '.add-progress-note', this.onNewProgressNoteClicked.bind(this));
+      this.$el.on('click', '.cancel-progress-note', this.onCancelNewProgressNote.bind(this));
+      this.$el.on('ajax:success', '.new-progress-note-form', this.onNewProgressNoteSaveSucceeded.bind(this));
+      this.$el.on('ajax:error', '.new-progress-note-form', this.onNewProgressNoteSaveFailed.bind(this));
+    },
+
+    onNewProgressNoteClicked: function(e) {
+      this.isAddingProgressNote = true;
+      this.render();
+    },
+
+    onCancelNewProgressNote: function(e) {
+      this.isAddingProgressNote = false;
+      this.render();
+    },
+
+    onSelectedIntervention: function (e) {
+      var interventionId = $(e.currentTarget).data('id');
+      this.selectedInterventionId = interventionId;
+      this.isAddingProgressNote = false;
+      this.isShowingNewIntervention = false;
+      this.render();
+    },
+
+    onAddNewIntervention: function (e) {
+      this.selectedInterventionId = null;
+      this.isShowingNewIntervention = true;
+      this.render();
+    },
+
+    onCancelNewIntervention: function (e) {
+      this.selectedInterventionId = this.defaultSelectedIntervention();
+      this.isShowingNewIntervention = false;
+      this.render();
+    },
+
+    onNewInterventionSaveSucceeded: function (e, data, status, xhr) {
+      this.interventions = [data].concat(this.interventions);
+      this.selectedInterventionId = this.defaultSelectedIntervention();
+      this.isShowingNewIntervention = false;
+      this.render();
+    },
+
+    onNewProgressNoteSaveSucceeded: function(e, data, status, xhr) {
+      var intervention = this.getSelectedIntervention();
+      intervention.progress_notes = intervention.progress_notes.concat([data]);
```

Comment:
FYI the sort order here is different between interventions and progress notes.  The interventions are recent->oldest while the progress notes are oldest->recent.  I think this makes sense with the way the page is laid out now, but something we could improve separately if we want.